core/schema/src/main/java/org/hawkular/metrics/schema/SchemaService.java

```diff
@@ -61,6 +66,62 @@ public void run(Session session, String keyspace, boolean resetDB) {
 
         session.execute("INSERT INTO " + keyspace + ".sys_config (config_id, name, value) VALUES " +
                 "('org.hawkular.metrics', 'version', '" + getNewHawkularMetricsVersion() + "')");
+
+        maybeUseTWCS(session, keyspace);
+    }
+
+    /**
+     * TWCS is not available in all versions of Cassandra on which we run. We attempt to update to TWCS, but if doing
+     * so fails, we just log a warning so that start up is not disrupted.
+     *
+     * See HWKMETRICS-608 and HWKMETRICS-603 for more info.
+     */
+    private void maybeUseTWCS(Session session, String keyspace) {
+        String cassandraVersion = getCassandraVersion(session);
+        String getCompactionClass;
+        ResultSet resultSet;
+
+        if (cassandraVersion.startsWith("2")) {
```

Comment:
I don't think so because this change is only being made in the release/0.18.0 branch. That version of metrics is compatible with Cassandra 3.x, Cassandra 2.2.x, and that's it.